app/connector/rest-swagger/src/main/java/io/syndesis/connector/rest/swagger/WithSyndesisHeaderFilterStrategy.java

```diff
@@ -52,7 +52,7 @@ public boolean applyFilterToCamelHeaders(final String headerName, final Object h
             // false | true | false => global has precedence over rest
             // true | false | true => global has precedence over rest
             // true | true | true => both agree header should be filtered out
-            return globalWouldFilter || globalWouldFilter && !restWouldFilter;
+            return globalWouldFilter || (globalWouldFilter && !restWouldFilter);
```

Comment:
there is something fishy here, this is equivalent to just return globalWouldFilter;